### PR TITLE
Ecs tools

### PR DESCRIPTION
**Pull request by mark-dce**
Tools to build a new cluster, add a load balancer, and deploy a docker-compose.yml of the complete DL stack.

### CODE REVIEW

File: .env.ecs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+S3CACHE_BUCKET_NAME=yale-image-samples/cantaloupe/cache
+PASSENGER_APP_ENV=development
+# Required by Postgres / Blacklight
+POSTGRES_HOST=localhost
+POSTGRES_DB=blacklight_yul_development
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=password
+# Point to an external manifests service
+IIIF_MANIFESTS_BASE_URL=/manifests/
+SOLR_URL=http://localhost:8983/solr/blacklight-development
```
Comment (maxkadel) [inline .env.ecs:12]: Is there a way to make the relationship between the `.env` file and the `.env.ecs` file clearer and less duplicative? I'm worried that they'll get out of sync in unhelpful ways.

I've separated the solr_url and solr_core in the management app, so I'll separate them here.

File: docker-compose.yml [hunk 2]
```diff
@@ -1,52 +1,76 @@
...
-      - iiif_manifest
-
-  iiif_image:
-    image: yalelibraryit/dc-iiif-cantaloupe:${CANTALOUPE_TAG}
-    env_file:
-      - .env
+      - '80:80'
+    logging:
+      driver: awslogs
+      options:
```
Comment (maxkadel) [inline docker-compose.yml:10]: I'm trying to figure out how to run this locally, and the logging is the current blocker. Even when I set the environment variables in the readme, I still get 
```bash
ERROR: for yul-dc-camerata_solr_1  Cannot create container for service solr: unknown log opt 'awslogs-stream-prefix' for awslogs log driver
```

File: docker-compose-simple.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+version: '3'
+services:
```
Comment (maxkadel) [inline docker-compose-simple.yml:2]: Is this file still needed?

File: bin/add-alb.sh [hunk 4]
```diff
@@ -0,0 +1,169 @@
+#!/bin/sh
+if [[ -z $1 ]]
```
Comment (mlooney) [inline bin/add-alb.sh:2]: this might be a job for getopts

Comment (mlooney) [reply bin/add-alb.sh:2]: also, quote your variables

File: .gitignore [hunk 5]
```diff
@@ -1 +1,5 @@
 git/
+ecs-params.yml
```
[the remark this answers is not in the document]
Comment (mark-dce) [reply .gitignore:2]: ecs-params.yml contains cluster specific information for it to work properly, so if we want a solution that can manage multiple clusters, we need a way to dynamically build a proper ecs-params.yml for each cluster dynamically.  There's a command called get-params.sh that reads the right configuration info from ECS and builds a file called `cluster-name-ecs-params.yml`.  Subsequent commands then run using `cluster-name-ecs-params.yml` instead of `ecs-params.yml`.  That let's anyone bootstrap any cluster in their environment without having stale configuration info checked into source control.

File: .env.ecs (hunk 1 above)
Comment (mark-dce) [reply .env.ecs]: Yes, we need to discuss and decide on conventions as a team about what goes where.  I think figuring out how to keep dev & prod as similar as possible with also help guide the solution.  In a future pass, I think we might be able to get away without needing .env.ecs.  For the first pass, I was trying to limit how much I had to change from what was in place already (I know, I didn't do a very good job - lots changed)

File: docker-compose.yml (hunk 2 above)
Comment (mark-dce) [reply docker-compose.yml:10]: There is no way to run it locally.  We need to get logging fixed - there's still a lot of refinement to be done here - think of this as shameless green.  But before adding too many more refinements, I want to make sure that this all works somewhere besides my development environment - it already feels like there's a lot of scaffolding here that the rest of the team hasn't been exposed to.

File: docker-compose-simple.yml (hunk 3 above)
Comment (mark-dce) [reply docker-compose-simple.yml:2]: Yes - one of the things I wanted was a known good configuration, so you can run `bin/deploy-simple-sh my-cluster-name` to validate that 1) your authentication and region are set correctly, 2) your build or params capture ran successfully, 3) and you can deploy a working container.  This gives us a known baseline even when the yale container relationships are in a broken state.

File: bin/add-alb.sh (hunk 4 above)
Comment (mark-dce) [reply bin/add-alb.sh:2]: Agreed - though I figured that bash might be more accessible to Yale's DevOps team.  So if they're going to maintain this long-run probably shell scripting; whereas, if Software engineering is going to own this, then ruby makes total sense (in honesty, it would have been faster to do in ruby and I could have gotten even better error detection in)
